crates/rpc/src/lib.rs

```diff
@@ -83,66 +79,93 @@ impl RpcServer {
     }
 
     /// Starts the HTTP-RPC server.
-    pub async fn run(self) -> Result<(ServerHandle, SocketAddr), anyhow::Error> {
-        const TEN_MB: u32 = 10 * 1024 * 1024;
-
-        let server = match self.ws_senders {
-				Some(_) => ServerBuilder::default(),
-				None => ServerBuilder::default().http_only(),
-			}
-            .max_connections(self.max_connections)
-            .max_request_body_size(TEN_MB)
-            .set_logger(self.logger)
-            .set_middleware(tower::ServiceBuilder::new()
-                .option_layer(self.cors)
-                .map_result(middleware::versioning::try_map_errors_to_responses)
-                .filter_async({
-                    let default_version = self.default_version;
-
-                    move |request: Request<Body>| async move {
-					// skip method_name checks for websocket handshake
-					if request.headers().get("sec-websocket-key").is_some() {
-						return Ok(request);
-					}
-
-                    middleware::versioning::prefix_rpc_method_names_with_version(request, TEN_MB, default_version).await
-                }})
-            )
-            .build(self.addr)
-            .await
-            .map_err(|e| match e {
-                jsonrpsee::core::Error::Transport(_) => {
-                    use std::error::Error;
-
-                    if let Some(inner) = e.source().and_then(|inner| inner.downcast_ref::<std::io::Error>()) {
-                        if let std::io::ErrorKind::AddrInUse = inner.kind() {
-                            return anyhow::Error::new(e)
-                                .context(format!("RPC address is already in use: {}.
-
-Hint: This usually means you are already running another instance of pathfinder.
-Hint: If this happens when upgrading, make sure to shut down the first one first.
-Hint: If you are looking to run two instances of pathfinder, you must configure them with different http rpc addresses.", self.addr));
-                        }
-                    }
-
-                    anyhow::Error::new(e)
-                }
-                _ => anyhow::Error::new(e),
-            })?;
-        let local_addr = server.local_addr()?;
-
-        let module = crate::module::Module::new(self.context);
-        let module = v03::register_methods(module)?;
-        let module = v04::register_methods(module)?;
-        let module = pathfinder::register_methods(module)?;
-        let module = match &self.ws_senders {
-            Some(ws_senders) => websocket::register_subscriptions(module, ws_senders.clone())?,
-            None => module,
+    pub fn spawn(self) -> Result<(JoinHandle<anyhow::Result<()>>, SocketAddr), anyhow::Error> {
+        use axum::routing::{get, post};
+
+        // TODO: allow OPTIONS
```

Comment:
isn't this managed by tower cors-middleware? then the comment could be removed

Reply:
I was unsure -- if you think so then I'll remove it. CORS is not something I'm familiar with.

Reply:
Please do.